utils/merge/mergeSchemas.js

```diff
@@ -34,12 +34,31 @@ module.exports = function mergeSchemas(schema1, schema2) {
       },
     },
   ];
+  const propsToOverride = [
+    {
+      name: 'required',
+      action(first, second) {
+        if (!schema1.properties || !schema2.properties) return first;
+
+        const firstProperties = Object.keys(schema1.properties);
+        const secondProperties = Object.keys(schema2.properties);
+
+        const optionalOverride = firstProperties
+          .filter(property => secondProperties.includes(property))
+          .filter(property => first.includes(property) && (!second || !second.includes(property)));
```

Comment:
How do you think, should we perform all checks in `filter` when there is no `second` at all? I mean, maybe we can check for `!second` right after `schema1/2.properties` check and exit early

Reply:
if there is no second at all, it exactly means that we need to override (`firstRequired` -> `['someProp']`, `secondRequired` -> `undefined` => it means that `someProp` has to be optional)
On other comments - up to codewriter if it affects nothing. I changed it